core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java

```diff
@@ -217,6 +217,10 @@ protected Transformer resolveTransformer(DataType source, DataType result) throw
       readLock.unlock();
     }
 
+    if (results.isEmpty()) {
```

Comment:
what is this for? If the cache is remembering the fact that there's no such transformer, why would a subsequent lookup succeed when the previous logic has failed? If a new transformer is in fact found, why isn't it cached?